jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java

```diff
@@ -288,9 +288,13 @@ protected void doStop() throws Exception
         }
 
         // Close any un-executed jobs
-        while (!_jobs.isEmpty())
+        while (true)
         {
             Runnable job = _jobs.poll();
+            if (job == null)
+            {
+                break;
+            }
```

Comment:
Looks good, but make it a little more concise in this way:

```suggestion
            if (job == null)
                break;
```